### PR TITLE
Add IDE0160 and IDE0161.

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to https://github.com/dotnet/roslyn/issues/60546 and #28791.

Adds IDE0160 and IDE0161. Also removes the option `csharp_style_namespace_declarations` from rule IDE0055 since it belongs with IDE0160 and IDE0161.

### CODE REVIEW

File: docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+| **Rule ID**              | IDE0161                                |
+| **Title**                | Use file-scoped namespace              |
+| **Category**             | Style                                  |
+| **Subcategory**          | Language rules (namespace preferences) |
+| **Applicable languages** | C#                                     |
+| **Options**              | `csharp_style_namespace_declarations`  |
+
+## Overview
+
+These rules concern how namespaces are declared. For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md:41]: As stated in https://github.com/dotnet/roslyn/issues/60546#issuecomment-1086854845, it's unclear why this option is even necessary. And what happens if you set it to `file_scoped` and enable IDE0160 (use block scoped namespaces)?

**Comment by mavasani** [comment]
@CyrusNajmabadi can you please review?

**Review by Youssef1313 — CHANGES_REQUESTED** [review]
I think you need to add a redirection from `ide0160` and `ide0161` to `ide0160-ide0161`. The help link in roslyn for all rules is:

https://github.com/dotnet/roslyn/blob/b55ec310613097be69bc8beb4b34f810771d0fa2/src/Analyzers/Core/Analyzers/Helpers/DiagnosticHelper.cs#L258-L270

File: docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md (hunk 1 above)
Comment (Youssef1313) [reply docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md:41]: The option is used when creating a new file in Visual Studio to determine whether the new file uses file scoped or not. This is true even if both IDE0160 and IDE0161 are disabled.

File: .openpublishing.redirection.fundamentals.json [hunk 2]
```diff
@@ -88,6 +88,14 @@
             "source_path_from_root": "/docs/fundamentals/code-analysis/style-rules/ide0074.md",
             "redirect_url": "/dotnet/fundamentals/code-analysis/style-rules/ide0054-ide0074"
         },
+        {
+            "source_path_from_root": "/docs/fundamentals/code-analysis/quality-rules/ide0160.md",
+            "redirect_url": "/dotnet/core/deploying/single-file/warnings/ide0160-ide0161.md"
+        },
+        {
+            "source_path_from_root": "/docs/fundamentals/code-analysis/quality-rules/ide0161.md",
+            "redirect_url": "/dotnet/core/deploying/single-file/warnings/ide0160-ide0161.md"
```
Comment (Youssef1313) [inline .openpublishing.redirection.fundamentals.json:97]:
```suggestion
            "redirect_url": "/dotnet/core/deploying/single-file/warnings/ide0160-ide0161"
```

File: .openpublishing.redirection.fundamentals.json [hunk 3]
```diff
@@ -88,6 +88,14 @@
             "source_path_from_root": "/docs/fundamentals/code-analysis/style-rules/ide0074.md",
             "redirect_url": "/dotnet/fundamentals/code-analysis/style-rules/ide0054-ide0074"
         },
+        {
+            "source_path_from_root": "/docs/fundamentals/code-analysis/quality-rules/ide0160.md",
+            "redirect_url": "/dotnet/core/deploying/single-file/warnings/ide0160-ide0161.md"
```
Comment (Youssef1313) [inline .openpublishing.redirection.fundamentals.json:93]:
```suggestion
            "redirect_url": "/dotnet/fundamentals/code-analysis/style-rules/ide0160-ide0161"
```

File: .openpublishing.redirection.fundamentals.json [hunk 4]
```diff
@@ -88,6 +88,14 @@
...
+        },
+        {
+            "source_path_from_root": "/docs/fundamentals/code-analysis/quality-rules/ide0161.md",
+            "redirect_url": "/dotnet/core/deploying/single-file/warnings/ide0160-ide0161"
```
Comment (Youssef1313) [inline .openpublishing.redirection.fundamentals.json:97]:
```suggestion
            "redirect_url": "/dotnet/fundamentals/code-analysis/style-rules/ide0160-ide0161"
```

Comment (gewarren) [reply .openpublishing.redirection.fundamentals.json:97]: Oops this was really sloppy of me. Thanks @Youssef1313!

File: .openpublishing.redirection.fundamentals.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (gewarren) [reply .openpublishing.redirection.fundamentals.json]: Yes, my bad. I think it's all straightened out with my latest commit.

File: docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+| **Rule ID**              | IDE0161                                |
+| **Title**                | Use file-scoped namespace              |
+| **Category**             | Style                                  |
+| **Subcategory**          | Language rules (namespace preferences) |
+| **Applicable languages** | C#                                     |
+| **Options**              | `csharp_style_namespace_declarations`  |
+
+## Overview
+
+These rules concern [how namespaces are declared](../../../csharp/language-reference/keywords/namespace.md). For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.
```
Comment (IEvangelist) [inline docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md:41]:
```suggestion
These rules apply to [how namespaces are declared](../../../csharp/language-reference/keywords/namespace.md). For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.
```

File: docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+## Options
+
+The option value specifies whether namespace declarations should be block scoped or file scoped. By default, namespace declarations are block scoped. This option is used by Visual Studio to determine how namespaces are declared when new code files are added to projects. Visual Studio honors the option value even if both `IDE0160` and `IDE0161` are disabled.
```
Comment (IEvangelist) [inline docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md:45]:
```suggestion
The option value specifies whether namespace declarations should be block scoped or file scoped. By default, namespace declarations are block-scoped. This option is used by Visual Studio to determine how namespaces are declared when new code files are added to projects. Visual Studio honors the option value even if both `IDE0160` and `IDE0161` are disabled.
```

**Review by IEvangelist — APPROVED** [review]

File: docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md (hunk 5 above)
Comment (gewarren) [reply docs/fundamentals/code-analysis/style-rules/ide0160-ide0161.md:41]:
```suggestion
These rules apply to [namespace declarations](../../../csharp/language-reference/keywords/namespace.md). For `IDE0161` to report violations when block-scoped namespaces are used, you must set the associated option to `file_scoped`.
```

**Pull request merged**